dotnet-interactive.Tests/JupyterKernelSpecTests.cs

```diff
@@ -35,24 +35,43 @@ public async Task Returns_success_output_when_kernel_installation_succeeded()
             var kernelDir = CreateDirectory();
 
             var result = await kernelSpec.InstallKernel(kernelDir);
-            result.Error.Should().BeEmpty();
-            result.ExitCode.Should().Be(0);
+            using var scope = new AssertionScope();
+            result.Succeeded.Should().BeTrue();
 
             _kernelInstallations.Add(new DirectoryInfo(kernelDir.Name));
```

Comment:
I just noticed this pattern. Adding cleanup steps after assertions will prevent the cleanup from happening if the assertion fails.